app/controllers/users/phone_setup_controller.rb

```diff
@@ -90,5 +91,11 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def confirm_user_in_account_setup
+      return if user_fully_authenticated? && in_multi_mfa_selection_flow?
+      return unless MfaPolicy.new(current_user).two_factor_enabled?
+      redirect_to account_path
```

Comment:
Between this and `confirm_user_authenticated_for_2fa_setup`, trying to think of a scenario where it's not sufficient to just check for `in_multi_mfa_selection_flow?` ?

```suggestion
      redirect_to account_path unless in_multi_mfa_selection_flow?
```

Reply:
I think we could drop the `user_fully_authenticated?` part, but the other one I think needs to remain since most tests do not go through MFA selection :/